spark/src/main/scala/org/apache/spark/sql/delta/commands/ReorgTableHelper.scala

```diff
@@ -0,0 +1,111 @@
+import org.apache.hadoop.fs.{FileStatus, Path}
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.execution.datasources.parquet.{ParquetFileFormat, ParquetToSparkSchemaConverter}
+import org.apache.spark.sql.types.{AtomicType, StructField, StructType}
+import org.apache.spark.util.SerializableConfiguration
+
+trait ReorgTableHelper extends Serializable {
+  /**
+   * Determine whether `fileSchema` has any columns that has a type that differ from
+   * `tablePhysicalSchema`.
+   */
+  protected def fileHasDifferentTypes(
+      fileSchema: StructType,
+      tablePhysicalSchema: StructType): Boolean = {
+    SchemaMergingUtils.transformColumns(fileSchema, tablePhysicalSchema) {
+      case (_, StructField(_, fileType: AtomicType, _, _),
+      Some(StructField(_, tableType: AtomicType, _, _)), _) if fileType != tableType =>
+        return true
+      case (_, field, _, _) => field
+    }
+    false
+  }
+
+  /**
+   * Apply a filter on the list of AddFile to only keep the files that have their physical parquet
+   * schema that satisfies the given filter function.
+   *
+   * Note: Filtering happens on the executors: **any variable captured by `filterFileFn` must be
+   * Serializable**
+   */
+  protected def filterParquetFilesOnExecutors(
+      spark: SparkSession,
+      files: Seq[AddFile],
+      snapshot: Snapshot)(
+      filterFileFn: StructType => Boolean): Seq[AddFile] = {
+
+    val serializedConf = new SerializableConfiguration(snapshot.deltaLog.newDeltaHadoopConf())
+    val ignoreCorruptFiles = spark.sessionState.conf.ignoreCorruptFiles
+    val assumeBinaryIsString = spark.sessionState.conf.isParquetBinaryAsString
+    val assumeInt96IsTimestamp = spark.sessionState.conf.isParquetINT96AsTimestamp
+    val dataPath = new Path(snapshot.deltaLog.dataPath.toString)
+
+    import org.apache.spark.sql.delta.implicits._
+
+    files.toDF(spark).as[AddFile].mapPartitions { iter =>
+        filterParquetFiles(iter.toList, dataPath, serializedConf.value, ignoreCorruptFiles,
+          assumeBinaryIsString, assumeInt96IsTimestamp)(filterFileFn).toIterator
+    }.collect()
+  }
+
+  protected def filterParquetFiles(
+      files: Seq[AddFile],
+      dataPath: Path,
+      configuration: Configuration,
+      ignoreCorruptFiles: Boolean,
+      assumeBinaryIsString: Boolean,
+      assumeInt96IsTimestamp: Boolean)(
+      filterFileFn: StructType => Boolean): Seq[AddFile] = {
+    val nameToAddFileMap = generateCandidateFileMap(dataPath, files)
+
+    val fileStatuses = nameToAddFileMap.map { case (absPath, addFile) =>
+      new FileStatus(
+        /* length */ addFile.size,
+        /* isDir */ false,
+        /* blockReplication */ 0,
+        /* blockSize */ 1,
+        /* modificationTime */ addFile.modificationTime,
+        new Path(absPath)
+      )
+    }
+
+    val footers = DeltaFileOperations.readParquetFootersInParallel(
+      configuration,
+      fileStatuses.toList,
+      ignoreCorruptFiles)
```

Comment:
So this param will allow it to ignore corrupt files? Should we instead never ignore corrupt files? It could potentially be dangerous because a corrupt file might recover later on and become unreadable. 
Any modification we do on tables are potentially unsafe if we ignore corrupt files.

Reply:
Good catch, I updated it to always fail on corrupted files + added a test